File: api-express/src/controllers/pedido-controller.js
import { createPedidoSchema } from "../validations/pedido-validations.js";
import * as pedidoService from '../services/pedido-service.js'
import AppMessage from '../utils/app-message.js'

export default class PedidoController {
    static async create(req, res) {
        const payload = createPedidoSchema.parse(req.body);
        payload.idCliente = req.cliente.id;

        const result = await pedidoService.create(payload);
        return res.status(201).send(new AppMessage('Pedido criado com sucesso', 201, result));
    }
}

File: api-actix/src/controllers/pedido_controller.rs
use actix_web::{post, web, HttpResponse, Result, HttpRequest};
use serde_json::Value;
use crate::services::pedido_service::PedidoService;
use crate::utils::app_message::{success_response, ApiError};
use crate::middlewares::is_authenticated::get_cliente_from_request;
use crate::db::AppState;

#[post("")]
async fn create(
    app_state: web::Data<AppState>,
    payload: web::Json<Value>,
    req: HttpRequest
) -> Result<HttpResponse, ApiError> {
    let cliente = get_cliente_from_request(&req)?;
    let mut pedido_payload = payload.into_inner();

    if let Some(obj) = pedido_payload.as_object_mut() {
        obj.insert("idCliente".to_string(), Value::String(cliente.id));
    }

    let result = PedidoService::create(&app_state.db_pool, pedido_payload).await?;
    Ok(success_response("Pedido criado com sucesso", 201, result))
}
